### PR TITLE
CLDR 41 Update - list

### PR DESCRIPTION
**Pull request by gouniLee**
### Checklist

* [ ] At least one test case is included for this feature or bug fix.
* [x] `ReleaseNotes` has added or is not needed.
* [x] This PR has passed all of the test cases on `Nodejs` and `Chrome Browser`.
* [ ] This PR has passed all of the test cases on `QT/QML`.
* [ ] This is an API breaking change.
* [ ] Requires a major version change.

### Issue Resolved / Feature Added
* Fix `genlist.js` tool code 
  *  Remove promoteFormats() which in inappropriate cause it override fallback locale data (i.e. en-CA data overrides en/ data due to that line )
*  Re-generate `list.json` files and fixed related test cases which were wrong.

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)

### Links
[//]: # (Related issues, references)

### CODE REVIEW

File: tools/cldr/genlist.js [hunk 1]
```diff
@@ -321,14 +321,6 @@ console.log("\n\nMerging formats forward ...");
 
 aux.mergeFormats(localePatterns, localePatterns, []);
 
-console.log("\n\nPromoting sublocales ...");
```
Comment (ehoogerbeets) [inline tools/cldr/genlist.js:324]: Why is this removed?

Comment (gouniLee) [reply tools/cldr/genlist.js:324]: I realized the data have generated wrongly due to this line. 
It overrides fallback locale data. for example, en-CA data overrides en data which should maintain it.

**Review by ehoogerbeets — APPROVED** [review]
